### DISCUSSION TITLE
Include Ollama binaries?

### DISCUSSION THREAD

**Discussion by Osiris-Team** [amithkoujalgi/ollama4j#74] [Ideas]
Users tend to be pretty lazy, thus including the binaries or an installation script that is automatically ran before using the API would be great!

Making this opt-in would make sense I guess.

Maybe helpful: https://github.com/ollama/ollama/actions/runs/10647220797

**Comment by daguava** [comment]
The readme has a pretty good requirements section that gives exact commands for Ollama installation, I don't think it would be a good idea to begin bundling other project's binaries. If the user is too lazy to run the 1 command from the readme, maybe you don't want them as your user :) .

**Comment by Osiris-Team** [comment]
Yeah, but keep in mind that sometimes larger projects have multiple dependencies and If each of those all require user action it can get a lot quick.

We also do not need to bundle the binary directly if you dont want to bloat up the library, instead we can add code to download the required binary.

**Comment by amithkoujalgi** [comment]
I agree with @daguava. On the other hand, @Osiris-Team, your feature request also sounds reasonable. Could you help outline the setup steps and provide some ideas on how to make the installation process cater to different platforms? It would be great to ensure that users on Windows, macOS, and Linux have a smooth experience. Your insights would be valuable here!
